### PR TITLE
Update existing settings that use presets

### LINKED ISSUE #3861: get-addon-settings.js migrations for v1.23.0
There will be more than one migration this release, ~~so it's better to code them all at once~~.

- [x]  #3878 Users that had the `editorCount` setting of `project-info` set to true, need to have the new `block-count` addon enabled (#3795)
- [ ] TBD what should be done about users on the old TurboWarp Dark preset after the update (#3839)
- [x] #3882 / Migrate scratchr2 settings to dark-www (https://github.com/ScratchAddons/ScratchAddons/pull/3710#issuecomment-952652969)
- [ ] TBD what should be done about users with the old Experimental Dark preset after the update (#3710)

Discussion for how presets should migrate is happening on https://github.com/ScratchAddons/ScratchAddons/discussions/3855

We could also remove old migration code by now.
And related, I find it funny that some users tell me they see dango in my profile still. Maybe the addon should be removed or made non-functional by now.

### PR DESCRIPTION
**Pull request by apple502j**
Resolves #3861

### CODE REVIEW

**Comment by mxmou** [comment]
I think it would make sense to only check if settings related to the changed ones (such as accent background and workspace background, but not menu bar background, highlight color, or full screen background) have the same values as in the preset.

File: background/get-addon-settings.js [hunk 1]
```diff
@@ -19,45 +45,60 @@ chrome.storage.sync.get(["addonSettings", "addonsEnabled"], ({ addonSettings = {
         delete settings.editorCount;
         addonsEnabled["block-count"] = true;
       }
+      if (addonId === "editor-dark-mode") {
+        // Transition v1.22 to v1.23
+        // TurboWarp Dark preset changes:
+        updatePresetIfMatching(
+          manifest.presets.find((p) => p.id === "tw-dark"),
+          settings,
+          {
```
Comment (TheColaber) [inline background/get-addon-settings.js:54]: Listing the present settings here is probably a bad thing to do. You can get the data from the manifest.

**Review by TheColaber — CHANGES_REQUESTED** [review]

File: background/get-addon-settings.js (hunk 1 above)
Comment (WorldLanguages) [reply background/get-addon-settings.js:54]: Old presets are not available in the manifest.

**Comment by apple502j** [comment]
@mxmou That'll work for Experimental Dark but not TW Dark (where only one setting actually changed).

File: background/get-addon-settings.js (hunk 1 above)
Comment (TheColaber) [reply background/get-addon-settings.js:54]: Oh, I thought this was the new one

**Comment by WorldLanguages** [comment]
Tested

**Pull request merged**